spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,463 @@
+          .key("spark.kubernetes.operator.namespace")
+          .enableDynamicOverride(false)
+          .description("Namespace that operator is deployed within.")
+          .typeParameterClass(String.class)
+          .defaultValue("default")
+          .build();
+
+  public static final ConfigOption<String> OPERATOR_WATCHED_NAMESPACES =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.watchedNamespaces")
+          .enableDynamicOverride(true)
+          .description(
+              "Comma-separated list of namespaces that the operator would be watching for "
+                  + "Spark resources. If set to '*' or unset, operator would watch all namespaces.")
+          .typeParameterClass(String.class)
+          .defaultValue("default")
+          .build();
+
+  public static final ConfigOption<Boolean> TERMINATE_ON_INFORMER_FAILURE =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.terminateOnInformerFailure")
+          .enableDynamicOverride(false)
+          .description(
+              "Enable to indicate informer errors should stop operator startup. If "
+                  + "disabled, operator startup will ignore recoverable errors, "
+                  + "caused for example by RBAC issues and will retry "
+                  + "periodically.")
+          .typeParameterClass(Boolean.class)
+          .defaultValue(false)
+          .build();
+
+  public static final ConfigOption<Integer> RECONCILER_TERMINATION_TIMEOUT_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.reconciler.terminationTimeoutSeconds")
+          .enableDynamicOverride(false)
+          .description(
+              "Grace period for operator shutdown before reconciliation threads are killed.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(30)
+          .build();
+
+  public static final ConfigOption<Integer> RECONCILER_PARALLELISM =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.reconciler.parallelism")
+          .enableDynamicOverride(false)
+          .description(
+              "Thread pool size for Spark Operator reconcilers. Unbounded pool would be used if "
+                  + "set to non-positive number.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(50)
+          .build();
+
+  public static final ConfigOption<Long> RECONCILER_FOREGROUND_REQUEST_TIMEOUT_SECONDS =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.reconciler.foregroundRequestTimeoutSeconds")
+          .enableDynamicOverride(true)
+          .description(
+              "Timeout (in seconds) to for requests made to API server. this "
+                  + "applies only to foreground requests.")
+          .typeParameterClass(Long.class)
+          .defaultValue(30L)
+          .build();
+
+  public static final ConfigOption<Long> SPARK_APP_RECONCILE_INTERVAL_SECONDS =
```

Comment:
Can we keep the pattern consistently? For example, since the name is `spark.kubernetes.operator.reconciler`,
`SPARK_APP_RECONCILE_INTERVAL_SECONDS` -> `RECONCILER_SPARK_APP_INTERVAL_SECONDS`?